scripts/tests/py/metadata.py

```diff
@@ -0,0 +1,225 @@
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+import re
+import yaml
+from typing import Optional
+from typing import Dict
+from typing import List
+from typing import Union
+
+@dataclass
+class Metadata:
+    py_script_path: Optional[str] = None
+    run: Optional[str] = None
+    app: Optional[str] = None
+    factoryreset: bool = False
+    comissioning_method: Optional[str] = None
+    discriminator: Optional[int] = None
+    kvs: Optional[str] = None
+    storage_path: Optional[str] = None
+    on_network_commission: Optional[str] = None
+    passcode: Optional[int] = None
+    endpoint: Optional[str] = None
+    manual_code: Optional[str] = None
+    PICS: Optional[str] = None
+    tests: Optional[str] = None
+
+    def copy_from_dict(self,attr_dict: Dict[str, str]) -> None:
+        """
+        Sets the value of the attributes from a dictionary.
+
+        Attributes:
+
+        attr_dict:
+          Dictionary that stores attributes value that should
+          be transferred to this class.
+        """
+        
+        if "app" in attr_dict:
+            self.app = attr_dict["app"]
+
+        if "run" in attr_dict:
+            self.run = attr_dict["run"]
+
+        if "discriminator" in attr_dict:
+            self.discriminator = attr_dict["discriminator"]
+
+        if "passcode" in attr_dict:
+            self.passcode = attr_dict["passcode"]
+
+        if "py_script_path" in attr_dict:
+            self.py_script_path=attr_dict["py_script_path"]
+
+        # TODO - set other attributes as well
+
+class MetadataReader:
+    """
+    A class to parse run arguments from the test scripts and 
+    resolve them to environment specific values.
+
+    Attributes:
+    
+    env: str
+       A dictionary that reprsents a environment configuration in
```

Comment:
```suggestion
       A dictionary that represents a environment configuration in
```

Also above, given that the type is "dictionary" the declaration of `:str` seems wrong.
Do we need these attributes to be public?